examples/tv-casting-app/linux/main.cpp

```diff
@@ -185,6 +195,71 @@ void InitCommissioningFlow(intptr_t commandArg)
     }
 }
 
+void OnContentLauncherSuccessResponse(void * context)
+{
+    ChipLogProgress(AppServer, "ContentLauncher: Default Success Response");
+}
+
+void OnContentLauncherFailureResponse(void * context, uint8_t status)
+{
+    ChipLogProgress(AppServer, "ContentLauncher: Default Failure Response: 0x%02x", status);
+}
+
+void DeviceEventCallback(const DeviceLayer::ChipDeviceEvent * event, intptr_t arg)
+{
+    if (event->Type == DeviceLayer::DeviceEventType::kCommissioningComplete)
+    {
+        Server * server = &(chip::Server::GetInstance());
+
+        chip::NodeId tvNodeId             = chip::DeviceLayer::Internal::DeviceControlServer::DeviceControlSvr().GetPeerNodeId();
+        chip::FabricIndex peerFabricIndex = chip::DeviceLayer::Internal::DeviceControlServer::DeviceControlSvr().GetFabricIndex();
+
+        chip::FabricInfo * fabric = server->GetFabricTable().FindFabricWithIndex(peerFabricIndex);
+        if (fabric == nullptr)
+        {
+            ChipLogError(AppServer, "Did not find fabric for index %d", peerFabricIndex);
+            return;
+        }
+
+        chip::DeviceProxyInitParams initParams = {
+            .sessionManager = &(server->GetSecureSessionManager()),
+            .exchangeMgr    = &(server->GetExchangeManager()),
+            .idAllocator    = &(server->GetSessionIDAllocator()),
+            .fabricInfo     = fabric,
+            // TODO: Determine where this should be instantiated
+            .imDelegate = chip::Platform::New<chip::Controller::DeviceControllerInteractionModelDelegate>(),
+        };
+
+        PeerId peerID = fabric->GetPeerId();
+        peerID.SetNodeId(tvNodeId);
+
+        chip::OperationalDeviceProxy * operationalDeviceProxy =
+            chip::Platform::New<chip::OperationalDeviceProxy>(initParams, peerID);
+        if (operationalDeviceProxy == nullptr)
+        {
+            ChipLogError(AppServer, "Failed in creating an instance of OperationalDeviceProxy");
+            return;
+        }
+
+        SessionHandle handle = server->GetSecureSessionManager().FindSecureSessionForNode(tvNodeId);
```

Comment:
The API docs for this function say it's a temporary solution for some tests.  Please figure out what the right API is to use here, but this one can't be right because it's based on NodeId, not PeerId.

Reply:
Opened an issue to track deprecation of FindSecureSessionForNode: https://github.com/project-chip/connectedhomeip/issues/13044

Reply:
The point is that we should not be adding new uses of this function at all and making the tech debt worse.  Have you talked to someone who knows this area (@pan-apple or @erjiaqing) about what the right APIs are to use here?

Reply:
OK, I checked with @pan-apple and he suggested `server->GetCASESessionManager()->FindOrEstablishSession()` as the right tool to use here.

Reply:
Alright, so FindOrEstablishSession does not find sessions we're the responder for, as discussed, and it sounds like the DNS-SD bits in these apps are pretty broken.....

I guess I'm OK with this if there is a clear timeframe commitment for when the DNS-SD bits will be fixed and this workaround removed.  What is that timeframe?